### PR TITLE
[release/2.7] enable NHWC batchnorm by default on ROCm7.0+

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
NHWC batchnorm enabled by default if ROCm>=7.0

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
! cherry-pick --onto rocm6.5_internal_testing

**Comment by okakarpa** [comment]
Created branch [autogenerated/rocm6.5_internal_testing_cherry-pick_pr-2180](https://github.com/ROCm/pytorch/tree/autogenerated/rocm6.5_internal_testing_cherry-pick_pr-2180) and https://github.com/ROCm/pytorch/pull/2207
